kbase-extension/static/kbase/js/common/cellComponents/fieldTableCellWidget.js

```diff
@@ -162,20 +174,43 @@ define([
          */
         function validateField(message) {
             // always clear the existing message to start with
-            clearMessage();
             switch (message.diagnosis) {
                 case 'required-missing':
                 case 'invalid':
+                    state.isValid = false;
                     showMessage(MESSAGE.error, message.errorMessage);
                     break;
                 case 'suspect':
+                    state.isValid = true;
                     showMessage(MESSAGE.warning, message.shortMessage);
                     break;
                 default:
+                    state.isValid = true;
+                    clearMessage(messageBaseClass);
                     break;
             }
         }
 
+        function setDuplicateValue(rows) {
+            state.isDuplicate = true;
+            let message = 'duplicate value found';
+            if (rows) {
+                message += ' on row';
```

Comment:
Grammar rules. I bet there's a better way to do this.
Sets the cases:
"on row 1",
"on rows 1 and 2",
"on rows 1, 2, and 3"

Reply:
There is a function `arrayToEnglish` in the `util/string` module that does adds the commas and the 'and'. It doesn't do that vital plural, though.

Reply:
should the "value" in "duplicate value found" also be pluralised?

Reply:
Yeah, if there's multiple rows. I'll fix that.

Reply:
Fixed! Thanks for the reminder about `arrayToEnglish`.